### PR TITLE
researched/updated some SpellAttribute flags

### PR DESCRIPTION
**Pull request by schell244**
I did some research and updated some spell dbc flags. 

1) converted spell.dbc (5875) to sql (using WDBX Editor) in order to properly filter, etc...
2) searched for spells ids which only had one certain flag (=SpellAttributes) 
3) put spell id in wow head: https://classic.wowhead.com/spell=XXX
4) wowhead did display what the flag stands for (in most cases)
5) converted flag from decimal to hex and compared what VMangos had assigned to it

### Todo / Checklist

Newly added flags need to be tested/confirmed:

SPELL_ATTR_PERIODICALLY_TRIGGER   
SPELL_ATTR_EX_STOPS_AUTO_ATTACK 
~~SPELL_ATTR_EX2_MODIFY_THREAD~~

### CODE REVIEW

**Comment by Oroxzy** [comment]
maybe rename SPELL_ATTR_ON_NEXT_SWING_1 to SPELL_ATTR_ON_NEXT_SWING then?

**Comment by schell244** [comment]
I will (re-)check SPELL_ATTR_ON_NEXT_SWING_1 later - when it's only for players next swing, then I would consider SPELL_ATTR_ON_NEXT_PLAYER_SWING

File: src/game/Spells/SpellDefines.h [hunk 1]
```diff
@@ -878,7 +878,7 @@ enum SpellAttributesEx4
     SPELL_ATTR_EX4_IGNORE_RESISTANCES         = 0x00000001,            // 0 From TC 3.3.5, but not present in 1.12 native DBCs. Add it with spell_mod to prevent a spell from being resisted.
     SPELL_ATTR_EX4_UNK1                       = 0x00000002,            // 1 proc on finishing move?
     SPELL_ATTR_EX4_REAL_TIME_DURATION         = 0x00000004,            // 2 aura continues to expire while player is offline
-    SPELL_ATTR_EX4_UNK3                       = 0x00000008,            // 3
+    SPELL_ATTR_EX4_UNK3                       = 0x00000008,            // 3 Uses resource: Energy? / Rouge only?
```
Comment (ratkosrb) [inline src/game/Spells/SpellDefines.h]: Rogue not Rouge

File: src/game/Spells/SpellDefines.h [hunk 2]
```diff
@@ -829,7 +829,7 @@ enum SpellAttributesEx2
     SPELL_ATTR_EX2_IS_ARCANE_CONCENTRATION    = 0x00800000,            // 23 Only mage Arcane Concentration have this flag
     SPELL_ATTR_EX2_UNK24                      = 0x01000000,            // 24
     SPELL_ATTR_EX2_UNK25                      = 0x02000000,            // 25
-    SPELL_ATTR_EX2_MODIFY_THREAD              = 0x04000000,            // 26 Almost all Threat, Taunt spells have this flag - should be unaffected by school immunity!?
+    SPELL_ATTR_EX2_MODIFY_THREAD              = 0x04000000,            // 26 modifies thread - not 100% sure (unaffected by school immunity)
```
Comment (ratkosrb) [inline src/game/Spells/SpellDefines.h]: Threat not Thread

**Comment by ulfgebhardt** [comment]
@schell244 @ratkosrb Whats the status on this? <3

**Comment by schell244** [comment]
Currently this PR is just renaming. The newly added flags are not proved to be correct yet.
I did some experiments with SPELL_ATTR_PERIODICALLY_TRIGGER https://github.com/vmangos/core/compare/development...schell244:SPELL_ATTR_PERIODICALLY_TRIGGER and I think this flag is correct, but for properly implementing & testing it, there is a lot of dangerous refactoring required (and I'm still a noob in c++). 

I have provided a similar PR with those new flags for CMangos as well, so I guess it's now waiting for someone who has time and knowledge to take the first steps :)
https://github.com/cmangos/mangos-classic/pull/438

**Comment by schell244** [comment]
I was not aware of the existence of `https://wow.tools/dbc/` where the actual wow development spell flags (including description) can be found.
https://github.com/cmangos/mangos-classic/pull/438#issuecomment-904500096
So changes of this PR are superfluous.
@ratkosrb feel free to close this PR (or leave it open as a remider for the implementation like CMangos)

**Comment by schell244** [comment]
https://github.com/vmangos/core/pull/1493

**Pull request closed without merging**